modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ModelBuilder/RightSidebar/RightSidebarObjectRelationshipDetails.tsx

```diff
@@ -197,20 +197,32 @@ export function RightSidebarObjectRelationshipDetails({
 						}
 					);
 
-					newObjectRelationship = {
-						...element.data,
-						deletionType: objectRelationship.deletionType,
-						label:
-							selfObjectRelationships &&
-							selfObjectRelationships?.length > 1
-								? selfObjectRelationships.length.toString()
-								: getLocalizableLabel(
-										defaultLanguageId,
-										objectRelationship.label,
-										objectRelationship.name
-								  ),
-						selfObjectRelationships: newSelfObjectRelationships,
-					};
+					if (
+						(element as Edge<ObjectRelationshipEdgeData>).data
+							?.objectRelationshipId === objectRelationship?.id
+					) {
+						newObjectRelationship = {
+							...element.data,
+							deletionType: objectRelationship.deletionType,
+							label:
+								isSelfObjectRelationship &&
+								selfObjectRelationships &&
+								selfObjectRelationships?.length > 1
```

Comment:
The optional chaining (?.) operator is not necessary here because you already checked for `selfObjectRelationships` in the line before. (same line as before)